javascript/javascript1/week4/homework/voiceA.js

```diff
@@ -0,0 +1,112 @@
+  const startIndexFirst = lowerCaseInput.indexOf(suffixFirst); //   console.log(match);
+  const startIndexThird = lowerCaseInput.indexOf(suffixThird);
+  const startIndexFourth = lowerCaseInput.indexOf(suffixFourth);
+  const startIndexSeventh = lowerCaseInput.indexOf(suffixSeventh);
+  if (lowerCaseInput.includes(suffixFirst)) {
+    if (startIndexFirst !== -1) {
+      const name = command
+        .slice(startIndexFirst + suffixFirst.length, command.length)
+        .trim();
+
+      if (nameSave.toLowerCase() === name.toLowerCase()) {
+        console.log(`Nice to see you again ${nameSave}`);
+      } else {
+        nameSave = name;
+        console.log(`Nice to meet you ${name}`);
+      }
+    }
+  }
+  if (lowerCaseInput == suffixSecond) {
+    if (nameSave === "") {
+      console.log("You havent put any name");
+    } else {
+      console.log(`Your name is ${nameSave}`);
+    }
+  }
+  if (lowerCaseInput.includes(suffixThird)) {
+    let listItem = command.slice(
+      startIndexThird + suffixThird.length + 1,
+      command.length - prefixThird.length - 1
+    );
+    listItem = listItem.charAt(0).toUpperCase() + listItem.slice(1);
+    todos.push(listItem);
+    console.log(`${listItem}added to the to-do`);
+  }
+  if (lowerCaseInput.includes(suffixFourth)) {
+    let removeItem = command.slice(
+      startIndexFourth + suffixFourth.length + 1,
+      command.length - prefixFourth.length - 1
+    );
+    removeItem = removeItem.charAt(0).toUpperCase() + removeItem.slice(1);
+    let removeIndex = todos.indexOf(removeItem);
+    todos.splice(removeIndex, 1);
+    console.log(`${removeItem}removed from the todo`);
+  }
+  if (lowerCaseInput.includes(suffixFifth)) {
+    console.log(todos);
+  }
+  if (lowerCaseInput.includes(suffixSixth)) {
+    let date = new Date();
+    let options = {
+      year: "numeric",
+      month: "long",
+      day: "numeric",
+    };
+    console.log(date.toLocaleDateString("en-GB", options));
+  }
+  if (lowerCaseInput.includes(suffixSeventh)) {
+    let remainingText = command.slice(
+      startIndexSeventh + suffixSeventh.length + 1,
+      command.length
+    );
+    remainingText = remainingText.split(" ");
+    console.log(remainingText);
+    if (remainingText[1] == "+" || remainingText[1] == "plus") {
+      result = Number(remainingText[0]) + Number(remainingText[2]);
+      console.log(result);
+    } else if (remainingText[1] == "*" || remainingText[1] == "multiple") {
+      result = Number(remainingText[0]) * Number(remainingText[2]);
+      console.log(result);
+    } else if (remainingText[1] == "/" || remainingText[1] == "divide") {
+      result = Number(remainingText[0]) / Number(remainingText[2]);
+      console.log(result);
+    } else if (remainingText[1] == "-" || remainingText[1] == "minus") {
+      result = Number(remainingText[0]) - Number(remainingText[2]);
+      console.log(result);
+    } else {
+      console.log("error");
+    }
```

Comment:
I would consider making this into an object like you did with the patterns.

An object like this and then return the value depending on the operator:

const operations = {
    "+": (a, b) => a + b,
    "-": (a, b) => a - b,
    "*": (a, b) => a * b,
    "/": (a, b) => a / b,
  };